pkg/controller/backup_controller.go

```diff
@@ -55,18 +57,22 @@ const backupVersion = 1
 type backupController struct {
 	*genericController
 
-	backupper             backup.Backupper
-	pvProviderExists      bool
-	lister                listers.BackupLister
-	client                arkv1client.BackupsGetter
-	clock                 clock.Clock
-	backupLogLevel        logrus.Level
-	newPluginManager      func(logrus.FieldLogger) plugin.Manager
-	backupTracker         BackupTracker
-	backupLocationLister  listers.BackupStorageLocationLister
-	defaultBackupLocation string
-	metrics               *metrics.ServerMetrics
-	newBackupStore        func(*api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error)
+	backupper                    backup.Backupper
+	pvProviderExists             bool
+	lister                       listers.BackupLister
+	client                       arkv1client.BackupsGetter
+	clock                        clock.Clock
+	backupLogLevel               logrus.Level
+	newPluginManager             func(logrus.FieldLogger) plugin.Manager
+	backupTracker                BackupTracker
+	backupLocationLister         listers.BackupStorageLocationLister
+	defaultBackupLocation        string
+	volumeSnapshotLocationLister listers.VolumeSnapshotLocationLister
+	volumeSnapshotLocations      []string
```

Comment:
Maybe `defaultSnapshotLocations`?  Also, I think it might be easier if this was a `map[string]string`, with provider names as keys, and VSL names as values? And the `NewBackupController` func can just require it to be passed in as a map (which #864 can do)